malcolm/modules/ADCore/util.py

```diff
@@ -92,3 +94,11 @@ def __init__(self,
         self.sourceType = ASourceTypes(sourceType)
         self.dataType = ADataTypes(dataType)
         self.datasetType = AAttributeTypes(datasetType)
+
+
+class XmlLayout:
+
+    @staticmethod
+    def get_layout_filename(file_dir, mri, suffix="attributes"):
```

Comment:
```suggestion
def make_xml_filename(file_dir, mri, suffix="attributes"):
    """Return a Block-specific filename for attribute or layout XML file"""
```
I'd dispense with the class, and just make this a plain old function